### PR TITLE
Eliminate dependency of pre-existing Objects for Postgres Operator install

### PR DESCRIPTION
**Pull request by jkatz**
The `pgo-backrest-repo-config` Secret consists of several default
files that can be generated when the Operator is initially loaded
(and reconciled on initialization as well). This removes additional
manual steps required when deploying the Operator through methods
such as OLM, making it easier to get started.

This also bumps the Kubernetes libraries up to 0.19.4.

### CODE REVIEW

File: internal/operator/common.go [hunk 1]
```diff
@@ -360,6 +371,82 @@ func initializeControllerWorkerCounts() {
...
+
+	for _, key := range defaultBackrestRepoConfigKeys {
+		defaultKeysSet = defaultKeysSet && len(secret.Data[key]) != 0
+	}
+
+	if defaultKeysSet {
+		return nil
+	}
+
+	// otherwise, set the defaults
```
Comment (cbandy) [inline internal/operator/common.go:412]: This logic is already covered by the following loop and `isModified`.
```suggestion
	// set any missing defaults
```

File: docs/content/installation/other/operator-hub.md [hunk 2]
```diff
@@ -15,44 +15,14 @@ that is available in OperatorHub.io.
...
-```
-git clone -b v{{< param operatorVersion >}} https://github.com/CrunchyData/postgres-operator.git
-cd postgres-operator
-```
-
-### Secrets
-
-Configure pgBackRest for your environment. If you do not plan to use AWS S3 to store backups, you can omit
-the `aws-s3` keys below.
+If you do not plan to use AWS S3 to store backups and would like to have the keys available for every backup, you can create a Secret as described below:
```
Comment (cbandy) [inline docs/content/installation/other/operator-hub.md:22]:
```suggestion
If you plan to use AWS S3 to store backups and would like to have the keys available for every backup, you can create a Secret as described below:
```

File: installers/ansible/roles/pgo-operator/tasks/main.yml [hunk 3]
```diff
@@ -266,13 +266,10 @@
...
           command: |
             {{ kubectl_or_oc }} create secret generic pgo-backrest-repo-config \
-              --from-file=config='{{ role_path }}/files/pgo-backrest-repo/config' \
-              --from-file=sshd_config='{{ role_path }}/files/pgo-backrest-repo/sshd_config' \
-              --from-file=aws-s3-ca.crt='{{ role_path }}/files/pgo-backrest-repo/aws-s3-ca.crt' \
               --from-literal=aws-s3-key='{{ backrest_aws_s3_key }}' \
               --from-literal=aws-s3-key-secret='{{ backrest_aws_s3_secret }}' \
               -n {{ pgo_operator_namespace }}
-          when: pgo_backrest_repo_config_result.rc == 1
+          when: pgo_backrest_repo_config_result.rc == 1 and ((backrest_aws_s3_key is defined and backrest_aws_s3_key != '') or (backrest_aws_s3_secret is defined and backrest_aws_s3_secret != ''))
```
[the remark this answers is not in the document]
Comment (jkatz) [reply installers/ansible/roles/pgo-operator/tasks/main.yml:272]: Yeah that's much easier to read.

File: internal/operator/common.go (hunk 1 above)
Comment (jkatz) [reply internal/operator/common.go:412]: Oh oops. Will adjust.

File: docs/content/installation/other/operator-hub.md (hunk 2 above)
Comment (jkatz) [reply docs/content/installation/other/operator-hub.md:22]: I like it that way.

**Review by cbandy — APPROVED** [review]

File: installers/olm/description.upstream.md [hunk 4]
```diff
@@ -49,20 +49,12 @@ export PGO_OPERATOR_NAMESPACE=pgo
 kubectl create namespace "$PGO_OPERATOR_NAMESPACE"
 ```
 
-### Secrets
+### Secrets (optional)
 
-Configure pgBackRest for your environment. If you do not plan to use AWS S3 to store backups, you can omit
-the `aws-s3` keys below.
+If you plan to use AWS S3 to store backups, you can configure your environment to automatically provide you AWS S3 credentials to all newly created PostgreSQL cluster:
```
Comment (andrewlecuyer) [inline installers/olm/description.upstream.md:54]: Looks like should be "...provide _your_ AWS S3 credentials...

And then also "clusters" at the end, e.g. "...newly created PostgreSQL clusters"

File: deploy/deploy.sh [hunk 5]
```diff
@@ -46,12 +46,12 @@ fi
 pgbackrest_aws_s3_key=$(awsKeySecret "aws-s3-key")
 pgbackrest_aws_s3_key_secret=$(awsKeySecret "aws-s3-key-secret")
 
-$PGO_CMD --namespace=$PGO_OPERATOR_NAMESPACE create secret generic pgo-backrest-repo-config \
-	--from-file=config=${PGO_CONF_DIR}/pgo-backrest-repo/config \
-	--from-file=sshd_config=${PGO_CONF_DIR}/pgo-backrest-repo/sshd_config \
-	--from-file=aws-s3-ca.crt=${PGO_CONF_DIR}/pgo-backrest-repo/aws-s3-ca.crt \
-	--from-literal=aws-s3-key="${pgbackrest_aws_s3_key}" \
-	--from-literal=aws-s3-key-secret="${pgbackrest_aws_s3_key_secret}"
+if [[ ! -z $pgbackrest_aws_s3_key ]] || [[ ! -z $pgbackrest_aws_s3_key_secret ]]
```
Comment (andrewlecuyer) [inline deploy/deploy.sh:49]: Another option here is `-n` to avoid the double negative.

File: installers/olm/description.openshift.md [hunk 6]
```diff
@@ -56,20 +56,12 @@ edit `conf/postgres-operator/pgo.yaml` and set `DisableFSGroup` to `true`.
 
 [Security Context Constraint]: https://docs.openshift.com/container-platform/latest/authentication/managing-security-context-constraints.html
 
-### Secrets
+### Secrets (optional)
 
-Configure pgBackRest for your environment. If you do not plan to use AWS S3 to store backups, you can omit
-the `aws-s3` keys below.
+If you plan to use AWS S3 to store backups, you can configure your environment to automatically provide you AWS S3 credentials to all newly created PostgreSQL cluster:
```
Comment (andrewlecuyer) [inline installers/olm/description.openshift.md:61]: Same as comment for `description.upstream.md`.

File: deploy/deploy.sh (hunk 5 above)
Comment (jkatz) [reply deploy/deploy.sh:49]: Noted for the future.

**Pull request merged**